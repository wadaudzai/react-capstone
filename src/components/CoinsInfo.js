import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import OneInfo from './OneInfo';
import '../styles/CoinsInfo.css';
import { fetchCoinsInfo } from '../redux/infoReducer';

const CoinsInfo = () => {
  const coinsInfo = useSelector((state) => state.coinsInfo);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCoinsInfo());
  }, []);

  return (
    <div>
      <OneInfo Info={coinsInfo} />
    </div>
  );
};
export default CoinsInfo;
