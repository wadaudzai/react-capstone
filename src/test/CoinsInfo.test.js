import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CoinsInfo from '../components/CoinsInfo';
import store from '../redux/configureStore';

it('Test CoinsInfo', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CoinsInfo />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });