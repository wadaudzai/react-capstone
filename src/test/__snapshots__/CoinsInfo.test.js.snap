// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CoinsInfo 1`] = `
<div>
  <div
    className="coin-info-super"
  >
    <div
      className="coin-title"
    >
      <h2>
        (
        )
      </h2>
    </div>
    <div
      className="prices-holder"
    >
      <h3
        className="price-text"
      >
        Coin Prices
      </h3>
      <div
        className="price"
      >
        <h3>
          Market Price:
        </h3>
        <p />
      </div>
      <div
        className="price"
      >
        <h3>
          Supply:
        </h3>
        <p />
      </div>
      <div
        className="price"
      >
        <h3>
          Market Cap:
        </h3>
        <p />
      </div>
      <div
        className="price"
      >
        <h3>
          Volume 24HRS:
        </h3>
        <p />
      </div>
      <div
        className="price"
      >
        <h3>
          Wrap 24HRS:
        </h3>
        <p />
      </div>
    </div>
  </div>
</div>
`;
